code/modules/awaymissions/mission_code/ruins/telecomns.dm

```diff
@@ -0,0 +1,495 @@
+
+/obj/structure/telecomms_trap_tank/bullet_act(obj/item/projectile/P)
+	explode()
+
+/obj/structure/telecomms_trap_tank/proc/explode()
+	explosion(loc, 1, 4, 6, flame_range = 6)
+	qdel(src)
+
+/obj/structure/telecomms_doomsday_device
+	name = "turret"
+	desc = "Looks like the cover to a turret. Not deploying, however?"
+	icon = 'icons/obj/turrets.dmi'
+	icon_state = "turretCover"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+	anchored = TRUE
+	//Has someone stolen loot from the ruins core room? If they try to leave without killing dvorak, they are punished.
+	var/thief = FALSE
+
+/obj/structure/telecomms_doomsday_device/proc/start_the_party(ruin_cheese_attempted = FALSE)
+	invisibility = 90
+	var/obj/machinery/syndicatebomb/doomsday/kaboom = new /obj/machinery/syndicatebomb/doomsday(get_turf(src))
+	kaboom.icon_state = "death-bomb-active"
+	var/atom/flick_holder = new /atom/movable/porta_turret_cover(loc)
+	for(var/obj/structure/telecomms_trap_tank/TTT in urange(15, get_turf(src)))
+		TTT.explode()
+	flick_holder.layer = kaboom.layer + 0.1
+	flick("popup", flick_holder)
+	sleep(1 SECONDS)
+	for(var/obj/machinery/shieldgen/telecomms/shield in urange(15, get_turf(src)))
+		shield.shields_up()
+	if(ruin_cheese_attempted)
+		for(var/obj/machinery/door/airlock/A in urange(20, get_turf(src)))
+			A.unlock(TRUE) //Fuck your bolted open doors, you cheesed it.
+			A.close(override = TRUE)
+	for(var/area/A in urange(25, get_turf(src), areas = TRUE))
+		if(istype(A, /area/space))
+			continue
+		if(ruin_cheese_attempted)
+			A.burglaralert(src)
+		else if(!A.fire)
+			A.firealert(kaboom)
+	for(var/obj/effect/abstract/cheese_trap/CT in urange(15, get_turf(src)))
+		qdel(CT)
+	kaboom.activate()
+	kaboom.icon_state = "death-bomb-active" // something funny goes on with icons here
+	qdel(flick_holder)
+	qdel(src)
+
+/obj/machinery/syndicatebomb/doomsday
+	name = "\improper D.V.O.R.A.K's Doomsday Device"
+	icon_state = "death-bomb"
+	desc = "Nice to see that AI's are improving on the standard doomsday device. Good to have variety. Also probably a good idea to start running."
+	anchored = TRUE
+	timer_set = 100
+	payload = /obj/item/bombcore/doomsday
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | ACID_PROOF
+
+/obj/machinery/syndicatebomb/doomsday/singularity_act()
+	return // saves me headache later
+
+/obj/machinery/syndicatebomb/doomsday/ex_act(severity)
+	return // No.
+
+/obj/machinery/syndicatebomb/doomsday/screwdriver_act(mob/user, obj/item/I)
+	to_chat(user, "<span class='danger'>[src] is welded shut! You can't get at the wires!</span>")
+
+/obj/machinery/syndicatebomb/doomsday/Initialize(mapload)
+	. = ..()
+	var/turf/T = get_turf(src)
+	if(is_station_level(T.z))
+		log_debug("something tried to spawn a telecomms doomsday ruin bomb on the station, deleting!")
+		return INITIALIZE_HINT_QDEL
+
+/obj/item/bombcore/doomsday
+	name = "a supermatter charged bomb core"
```

Comment:
I don't think including `a` in the name is something we often do?

Reply:
removed